ch_pipeline/core/containers.py

```diff
@@ -942,6 +942,15 @@ def dataset_name_allowed(self, name: str) -> bool:
         parent_name, name = posixpath.split(name)
         return parent_name == "/" or self.group_name_allowed(parent_name)
 
+    @property
```

Comment:
I would consider caching this one. Either with the built-in `cached_property` decorator, or with the one we have in `caput.cache` if we think we want to support versions earlier than Python 3.8.

Reply:
Using the one from `caput` gave the following error while testing:
```
TypeError: unhashable type: 'HFBData'
```
I now used the one from `functools` and didn't see the error anymore.